src/storage/lock_manager.rs

```diff
@@ -9,6 +9,53 @@ pub struct Lock {
     pub hash: u64,
 }
 
+/// Time to wait for lock released when encountering locks.
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum WaitTimeout {
+    None,
+    Default,
```

Comment:
How about implementing the default trait for `WaitTimeout` instead? I think we do not need `WaitTimeout::Defaul`t, maybe we can use Millis(0) instead

Reply:
The idea of using an enum here is to make it explicit when we are using a default value and when we are using a specified value, making 0 mean default is making the difference implicit and therefore harder to understand.